RecoBTag/DeepFlavour/plugins/DeepFlavourExporter.cc

```diff
@@ -0,0 +1,133 @@
+
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/one/EDAnalyzer.h"
+
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "CommonTools/UtilAlgos/interface/TFileService.h"
+
+#include "DataFormats/PatCandidates/interface/Jet.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/Framework/interface/makeRefToBaseProdFrom.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/StreamID.h"
+
+#include "DataFormats/BTauReco/interface/JetTag.h"
+#include "DataFormats/DeepFormats/interface/DeepFlavourTagInfo.h"
+
+#include <algorithm>
+#include <string>
+
+#include "TTree.h"
+
+class DeepFlavourExporter : public edm::one::EDAnalyzer<edm::one::SharedResources> {
+
+  public:
+	  explicit DeepFlavourExporter(const edm::ParameterSet&);
+	  ~DeepFlavourExporter();
+
+	  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+  private:
+
+    typedef std::vector<reco::DeepFlavourTagInfo> TagInfoCollection;
+    typedef reco::JetTagCollection JetTagCollection;
+
+    virtual void beginJob() override;
+    virtual void analyze(const edm::Event&, const edm::EventSetup&) override;
+    virtual void endJob() override;
+
+	  const edm::EDGetTokenT< TagInfoCollection > tag_info_src_;
+	  edm::EDGetTokenT<edm::View<pat::Jet>> jet_src_;
+
+    // to keep data to be saved in the TTree
+    std::vector<std::string> disc_names_;
+    std::vector<float> disc_values_;
+    deep::DeepFlavourFeatures features_;
+    
+    bool use_jet_src_;
+
+    edm::Service<TFileService> fs;
```

Comment:
Do not hold onto the service handle as member data. Just request it when you need it.

Reply:
I will remove the Exporter plugin becuase it is not needed anymore

Reply:
Exporter removed with https://github.com/cms-sw/cmssw/pull/19893/commits/de7e130c318f4dc84a506c2e2aef7d32900013a8